blacksheep/server/application.py

```diff
@@ -608,3 +608,36 @@ async def __call__(self, scope, receive, send):
 
         request.scope = None  # type: ignore
         request.content.dispose()
+
+
+class ASGIApplication(Application):
+    def __init__(
+        self,
+        *,
+        router: Optional[Router] = None,
+        services: Optional[Container] = None,
+        debug: bool = False,
+        show_error_details: Optional[bool] = None,
+        mount_class: Optional[Mount] = None,
+    ):
+        super().__init__(
+            router=router,
+            services=services,
+            debug=debug,
+            show_error_details=show_error_details,
+        )
+        if mount_class is None:
+            self._mount = Mount()
```

Comment:
This is a mistake, the code is setting `self_mount` only if the input parameter is `None`.

You meant to do:

```python
if mount is None:
    mount = Mount()

self._mount: Mount = mount
```